src/components/Posts.context.js

```diff
@@ -0,0 +1,48 @@
+import React, { createContext, Component } from 'react'
+
+const PostsContext = createContext({
+    posts: [],
+    subreddit: '',
+    isFetching: false,
+    selectedSubreddit: () => {}
```

Comment:
I'm not sure about that name

```suggestion
        selectSubreddit: (subreddit) => this.fetchPosts(subreddit)
```